src/server/db_slice.cc

```diff
@@ -88,9 +88,13 @@ class PrimeEvictionPolicy {
   static constexpr bool can_gc = true;
 
   PrimeEvictionPolicy(const DbContext& cntx, bool can_evict, ssize_t mem_budget, ssize_t soft_limit,
-                      DbSlice* db_slice)
-      : db_slice_(db_slice), mem_budget_(mem_budget), soft_limit_(soft_limit), cntx_(cntx),
-        can_evict_(can_evict) {
+                      DbSlice* db_slice, bool apply_memory_limit)
+      : db_slice_(db_slice),
+        mem_budget_(mem_budget),
+        soft_limit_(soft_limit),
+        cntx_(cntx),
+        can_evict_(can_evict),
+        apply_memory_limit_(apply_memory_limit) {
```

Comment:
instead of passing apply_memory_limit you can pass `mem_budget` as INT64_MAX

Reply:
apply_memory_limit parameter is not needed

Reply:
there is a problem with this solution as we use the evp.memory_budget() to update db_slice.memory_budget_ so if we pass INT64_MAX we will mess with the db_slice.memory_budget_